test/files/run/t8549.scala

```diff
@@ -180,7 +180,7 @@ object Test extends App {
   // check(mutable.ArraySeq(1, 2, 3))(         "rO0ABXNyACFzY2FsYS5jb2xsZWN0aW9uLm11dGFibGUuQXJyYXlTZXEVPD3SKEkOcwIAAkkABmxlbmd0aFsABWFycmF5dAATW0xqYXZhL2xhbmcvT2JqZWN0O3hwAAAAA3VyABNbTGphdmEubGFuZy5PYmplY3Q7kM5YnxBzKWwCAAB4cAAAAANzcgARamF2YS5sYW5nLkludGVnZXIS4qCk94GHOAIAAUkABXZhbHVleHIAEGphdmEubGFuZy5OdW1iZXKGrJUdC5TgiwIAAHhwAAAAAXNxAH4ABQAAAAJzcQB+AAUAAAAD")
   check(mutable.AnyRefMap("a" -> "A"))(       "rO0ABXNyACJzY2FsYS5jb2xsZWN0aW9uLm11dGFibGUuQW55UmVmTWFwAAAAAAAAAAECAAdJAAVfc2l6ZUkAB192YWNhbnRJAARtYXNrTAAMZGVmYXVsdEVudHJ5dAARTHNjYWxhL0Z1bmN0aW9uMTtbACtzY2FsYSRjb2xsZWN0aW9uJG11dGFibGUkQW55UmVmTWFwJCRfaGFzaGVzdAACW0lbAClzY2FsYSRjb2xsZWN0aW9uJG11dGFibGUkQW55UmVmTWFwJCRfa2V5c3QAE1tMamF2YS9sYW5nL09iamVjdDtbACtzY2FsYSRjb2xsZWN0aW9uJG11dGFibGUkQW55UmVmTWFwJCRfdmFsdWVzcQB+AAN4cAAAAAEAAAAAAAAAB3NyADNzY2FsYS5jb2xsZWN0aW9uLm11dGFibGUuQW55UmVmTWFwJEV4Y2VwdGlvbkRlZmF1bHQAAAAAAAAAAQIAAHhwdXIAAltJTbpgJnbqsqUCAAB4cAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+UkA2AAAAAHVyABNbTGphdmEubGFuZy5PYmplY3Q7kM5YnxBzKWwCAAB4cAAAAAhwcHBwcHB0AAFhcHVxAH4ACQAAAAhwcHBwcHB0AAFBcA==")
   check(mutable.ArrayStack(1, 2, 3))(       "rO0ABXNyACNzY2FsYS5jb2xsZWN0aW9uLm11dGFibGUuQXJyYXlTdGFja3bdxXbcnLBeAgACSQAqc2NhbGEkY29sbGVjdGlvbiRtdXRhYmxlJEFycmF5U3RhY2skJGluZGV4WwAqc2NhbGEkY29sbGVjdGlvbiRtdXRhYmxlJEFycmF5U3RhY2skJHRhYmxldAATW0xqYXZhL2xhbmcvT2JqZWN0O3hwAAAAA3VyABNbTGphdmEubGFuZy5PYmplY3Q7kM5YnxBzKWwCAAB4cAAAAANzcgARamF2YS5sYW5nLkludGVnZXIS4qCk94GHOAIAAUkABXZhbHVleHIAEGphdmEubGFuZy5OdW1iZXKGrJUdC5TgiwIAAHhwAAAAA3NxAH4ABQAAAAJzcQB+AAUAAAAB")
-  check(mutable.DoubleLinkedList(1, 2, 3))( "rO0ABXNyAClzY2FsYS5jb2xsZWN0aW9uLm11dGFibGUuRG91YmxlTGlua2VkTGlzdI73LKsKRr1RAgADTAAEZWxlbXQAEkxqYXZhL2xhbmcvT2JqZWN0O0wABG5leHR0AB5Mc2NhbGEvY29sbGVjdGlvbi9tdXRhYmxlL1NlcTtMAARwcmV2cQB+AAJ4cHNyABFqYXZhLmxhbmcuSW50ZWdlchLioKT3gYc4AgABSQAFdmFsdWV4cgAQamF2YS5sYW5nLk51bWJlcoaslR0LlOCLAgAAeHAAAAABc3EAfgAAc3EAfgAEAAAAAnNxAH4AAHNxAH4ABAAAAANzcQB+AABwcQB+AAtxAH4ACXEAfgAHcQB+AANw")
```

Comment:
Serialization changed because we now emit a more precise type for the `next` field that's mixed in from a trait:

```
➜  scala git:(fields-wip) qsc -Xprint:mixin /Users/adriaan/Library/Preferences/IdeaIC2016.1/scratches/scratch.scala
[[syntax trees at end of                     mixin]] // scratch.scala
package <empty> {
  abstract trait Seq extends Object;
  abstract trait SeqLike extends Object;
  abstract trait LinkedListLike extends Object with SeqLike {
    <accessor> <sub_synth> def next(): Seq;
    @scala.runtime.TraitSetter <accessor> <sub_synth> def next_=(x$1: Seq): Unit;
    def /*LinkedListLike*/$init$(): Unit = {
      ()
    }
  };
  class LinkedList extends Object with Seq with LinkedListLike {
    override <accessor> def next(): LinkedList = LinkedList.this.next;
    private[this] var next: LinkedList = _;
    @scala.runtime.TraitSetter override <accessor> def next_=(x$1: LinkedList): Unit = LinkedList.this.next = x$1;
    override <bridge> <artifact> def next_=(x$1: Seq): Unit = LinkedList.this.next_=(x$1.$asInstanceOf[LinkedList]());
    override <bridge> <artifact> def next(): Seq = LinkedList.this.next();
    def <init>(): LinkedList = {
      LinkedList.super.<init>();
      LinkedList.super./*LinkedListLike*/$init$();
      LinkedList.this.next_=(this);
      ()
    }
  }
}

➜  scala git:(fields-wip) scalac -Xprint:mixin /Users/adriaan/Library/Preferences/IdeaIC2016.1/scratches/scratch.scala
[[syntax trees at end of                     mixin]] // scratch.scala
package <empty> {
  abstract trait Seq extends Object;
  abstract trait SeqLike extends Object;
  abstract trait LinkedListLike extends Object with SeqLike {
    <accessor> def next(): Seq = LinkedListLike.this.next();
    @runtime.TraitSetter <accessor> def next_=(x$1: Seq): Unit = LinkedListLike.this.next_=(x$1);
    def /*LinkedListLike*/$init$(): Unit = {
      ()
    }
  };
  class LinkedList extends Object with Seq with LinkedListLike {
    <accessor> def next(): Seq = LinkedList.this.next;
    private[this] var next: Seq = _;
    @runtime.TraitSetter <accessor> def next_=(x$1: Seq): Unit = {
      LinkedList.this.next = x$1;
      ()
    };
    def <init>(): LinkedList = {
      LinkedList.super.<init>();
      LinkedList.super./*LinkedListLike*/$init$();
      LinkedList.this.next_=(this);
      ()
    }
  }
}
```

Smaller example:
```
trait Seq[A]
trait SeqLike[A, +This <: SeqLike[A, This] with Seq[A]]
trait LinkedListLike[A, This <: Seq[A] with LinkedListLike[A, This]] extends SeqLike[A, This] {
  var next: This = _
}

class LinkedList[A]() extends Seq[A] with LinkedListLike[A, LinkedList[A]] {
  next = this
}
```